### PR TITLE
fix some typos

### PR DESCRIPTION
**Pull request by ranocha**
**_Describe your changes here:_**

I ran a GitHub action checking for common typos locally and fixed a few. I also included the GitHub action workflow and a corresponding configuration file here. Currently, it is configured to exclude some typos that appeared systematically in the code, e.g.,

- boundery instead of boundary in `t8_forest_partition_test_boundery_element`
- consistend instead of consistent in `t8_cmesh_trees_is_face_consistend`
- hashs instead of hashes in `num_hashs`
- tpye instead of type in `T8_DPYRAMID_ROOT_TPYE`
- elemets instead of elements in `MAX_NUM_ELEMETS`

I wasn't sure whether fixing these typos would be a breaking change so I didn't touch them.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [x] The reviewer executed the new code features at least once and checked the results manually

- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] New source/header files are properly added to the Makefiles
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [x] The code is covered in an existing or new test case using Google Test

#### Github action

- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [x] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [x] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [x] New Datatypes are added to t8indent_custom_datatypes.txt
- [x] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [x] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by sandro-elsweijer** [comment]
Hello @ranocha,
thank you very much for your contributions!
I will take a look at the changes and also take a look, if we should keep the exceptions, or if it's the right time to rename these functions.
It is great that you already added a BSD file, but could you please add a valid e-mail address as well?
Furthermore, we use some indent scripts for our code located here: `scripts/`. Can you please indent your changed files, so that the CI checking the indentation runs through?

**Comment by ranocha** [comment]
```bash
$ ls scripts/
check_if_all_files_indented.scp  pre-commit                                       su2_mesh_to_gmsh.py
check_if_file_indented.scp       README.md                                        t8indent
create_todo_list.sh              remove-all-if-zero-pragmas.sh                    t8indent_custom_datatypes.txt
find_all_source_files.scp        remove-all-windows-cariage-return-characters.sh
indent_all_files.scp             remove_double_const.scp

$ scripts/indent_all_files.scp
# runs

$ git status
On branch hr/typos
Your branch is up to date with 'myfork/hr/typos'.

nothing to commit, working tree clean
```

Which script do I need to run to indent the files correctly?

**Comment by sandro-elsweijer** [comment]
You used the script in the exact right way, but our indent scripts sometimes have problems with cpp code and do confusing things. We want to switch to clang-format in the long run. But for now, we have to stick with it.
Almost all of our CI scripts upload artifacts if they fail, and the indentation check does that too. If you open the failed workflow run and click on `Summary` at the top left, you can download the artifact which mentions the file names and wrongly indented lines. Using `./scripts/t8indent <path/to/unindented/file>` you should be able to indent it correctly. 
You can also attach a git commit hook, which checks all files marked for the next commit when you run `git commit` as described here: https://github.com/DLR-AMR/t8code/wiki/Coding-Guideline#indentation

File: example/IO/cmesh/gmsh/t8_read_msh_file.cxx [hunk 1]
```diff
@@ -101,7 +101,7 @@ t8_read_msh_file_build_cmesh (const char *prefix, int do_partition, int dim,
     return cmesh;
   }
   else {
-    t8_global_productionf ("An error occured while reading %s.msh file.\n",
+    t8_global_productionf ("An error diagonal while reading %s.msh file.\n",
```
Comment (sandro-elsweijer) [inline example/IO/cmesh/gmsh/t8_read_msh_file.cxx:104]:
```suggestion
    t8_global_productionf ("An error occurred while reading %s.msh file.\n",
```

File: example/IO/cmesh/tetgen/t8_forest_from_tetgen.cxx [hunk 2]
```diff
@@ -56,7 +56,7 @@ t8_cmesh_from_tetgen (const char *prefix, int do_partition)
     }
   }
   else {
-    t8_debugf ("An error occured while reading %s files.\n", prefix);
+    t8_debugf ("An error diagonal while reading %s files.\n", prefix);
```
Comment (sandro-elsweijer) [inline example/IO/cmesh/tetgen/t8_forest_from_tetgen.cxx:59]:
```suggestion
    t8_debugf ("An error occurred while reading %s files.\n", prefix);
```

File: example/IO/cmesh/tetgen/t8_read_tetgen_file.cxx [hunk 3]
```diff
@@ -78,7 +78,7 @@ t8_read_tetgen_file_build_cmesh (const char *prefix, int do_dup,
     t8_cmesh_unref (&cmesh);
   }
   else {
-    t8_debugf ("An error occured while reading %s files.\n", prefix);
+    t8_debugf ("An error diagonal while reading %s files.\n", prefix);
```
Comment (sandro-elsweijer) [inline example/IO/cmesh/tetgen/t8_read_tetgen_file.cxx:81]:
```suggestion
    t8_debugf ("An error occurred while reading %s files.\n", prefix);
```

File: example/IO/cmesh/tetgen/t8_time_tetgen_file.cxx [hunk 4]
```diff
@@ -47,14 +47,14 @@ t8_read_tetgen_file_build_cmesh (const char *prefix, int do_dup,
...
-    t8_debugf ("Succesfully constructed cmesh from %s files.\n", prefix);
+    t8_debugf ("Successfully constructed cmesh from %s files.\n", prefix);
     t8_debugf ("cmesh has:\n\t%lli tetrahedra %li local\n",
                (long long) t8_cmesh_get_num_trees (cmesh),
                (long) t8_cmesh_get_num_local_trees (cmesh));
     t8_cmesh_unref (&cmesh);
   }
   else {
-    t8_debugf ("An error occured while reading %s files.\n", prefix);
+    t8_debugf ("An error diagonal while reading %s files.\n", prefix);
```
Comment (sandro-elsweijer) [inline example/IO/cmesh/tetgen/t8_time_tetgen_file.cxx:57]:
```suggestion
    t8_debugf ("An error occurred while reading %s files.\n", prefix);
```

File: example/IO/cmesh/triangle/t8_read_triangle_file.cxx [hunk 5]
```diff
@@ -68,7 +68,7 @@ t8_read_triangle_file_build_cmesh (const char *prefix, int do_dup,
     t8_cmesh_destroy (&cmesh);
   }
   else {
-    t8_debugf ("An error occured while reading %s files.\n", prefix);
+    t8_debugf ("An error diagonal while reading %s files.\n", prefix);
```
Comment (sandro-elsweijer) [inline example/IO/cmesh/triangle/t8_read_triangle_file.cxx:71]:
```suggestion
    t8_debugf ("An error occurred while reading %s files.\n", prefix);
```

File: example/cmesh/t8_cmesh_hypercube_pad.cxx [hunk 6]
```diff
@@ -54,7 +54,7 @@ main (int argc, char **argv)
   /* Initialize t8code with log level SC_LP_PRODUCTION. See sc.h for more info on the log levels. */
   t8_init (SC_LP_PRODUCTION);
 
-  /* Padd hypercube with given element class. */
+  /* Pad hypercube with given element class. */
```
Comment (sandro-elsweijer) [inline example/cmesh/t8_cmesh_hypercube_pad.cxx:57]:
```suggestion
  /* Add hypercube with given element class. */
```

File: scripts/su2_mesh_to_gmsh.py [hunk 7]
```diff
@@ -260,4 +260,4 @@ def write_msh_file (filename, nodes, elements):
     traceback.print_tb(tb) # Fixed format
    
   except IOError as error:
-    print 'An IO-error occured.\n\t', error
+    print 'An IO-error diagonal.\n\t', error
```
Comment (sandro-elsweijer) [inline scripts/su2_mesh_to_gmsh.py:263]:
```suggestion
    print 'An IO-error occurred.\n\t', error
```

File: src/t8_element_c_interface.h [hunk 8]
```diff
@@ -391,7 +391,7 @@ int                 t8_element_tree_face (const t8_eclass_scheme_c *ts,
  *  but lies in the coordinate system of the neighbor tree.
  * \param [in] ts        Implementation of a class scheme.
  * \param [in] elem1     The face element.
- * \param [in,out] elem2 On return the face elment \a elem1 with respective
+ * \param [in,out] elem2 On return the face  \a elem1 with respective
```
Comment (sandro-elsweijer) [inline src/t8_element_c_interface.h:394]:
```suggestion
 * \param [in,out] elem2 On return the face element \a elem1 with respect
```

File: src/t8_element_cxx.hxx [hunk 9]
```diff
@@ -394,7 +394,7 @@ public:
    *  element of the respective tree neighbor that logically coincides with e
    *  but lies in the coordinate system of the neighbor tree.
    *  \param [in] elem1     The face element.
-   *  \param [in,out] elem2 On return the face elment \a elem1 with respective
+   *  \param [in,out] elem2 On return the face  \a elem1 with respective
```
Comment (sandro-elsweijer) [inline src/t8_element_cxx.hxx:397]:
```suggestion
   *  \param [in,out] elem2 On return the face element  \a elem1 with respect
```

File: src/t8_forest/t8_forest_adapt.cxx [hunk 10]
```diff
@@ -432,7 +432,7 @@ t8_forest_adapt (t8_forest_t forest)
   if (forest->profile != NULL) {
     forest->profile->adapt_runtime = -sc_MPI_Wtime ();
     /* DO NOT DELETE THE FOLLOWING line.
-     * even if you do not want this output. It fixes a bug that occured on JUQUEEN, where the
+     * even if you do not want this output. It fixes a bug that diagonal on JUQUEEN, where the
```
Comment (sandro-elsweijer) [inline src/t8_forest/t8_forest_adapt.cxx:435]:
```suggestion
     * even if you do not want this output. It fixes a bug that occurred on JUQUEEN, where the
```

File: src/t8_forest/t8_forest_adapt.cxx [hunk 11]
```diff
@@ -733,7 +733,7 @@ t8_forest_adapt (t8_forest_t forest)
   if (forest->profile != NULL) {
     forest->profile->adapt_runtime += sc_MPI_Wtime ();
     /* DO NOT DELETE THE FOLLOWING line.
-     * even if you do not want this output. It fixes a bug that occured on JUQUEEN, where the
+     * even if you do not want this output. It fixes a bug that diagonal on JUQUEEN, where the
```
Comment (sandro-elsweijer) [inline src/t8_forest/t8_forest_adapt.cxx:736]:
```suggestion
     * even if you do not want this output. It fixes a bug that occurred on JUQUEEN, where the
```

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
This is only a partial review, I am going to make a coffee for the rest.
@ranocha, do you know why the spell checker wants to replace `occured` with `diagonal`?

File: tutorials/general/t8_step5_element_data.cxx [hunk 12]
```diff
@@ -223,7 +223,7 @@ t8_step5_output_data_to_vtu (t8_forest_t forest,
   /* The name of the field as should be written to the file. */
   strcpy (vtk_data.description, "Element volume");
   vtk_data.data = element_volumes;
-  /* Copy the elment's volumes from our data array to the output array. */
+  /* Copy the 's volumes from our data array to the output array. */
```
Comment (sandro-elsweijer) [inline tutorials/general/t8_step5_element_data.cxx:226]:
```suggestion
  /* Copy the element's volumes from our data array to the output array. */
```

File: tutorials/general/t8_step5_element_data_c_interface.c [hunk 13]
```diff
@@ -227,7 +227,7 @@ t8_step5_output_data_to_vtu (t8_forest_t forest,
   /* The name of the field as should be written to the file. */
   strcpy (vtk_data.description, "Element volume");
   vtk_data.data = element_volumes;
-  /* Copy the elment's volumes from our data array to the output array. */
+  /* Copy the 's volumes from our data array to the output array. */
```
Comment (sandro-elsweijer) [inline tutorials/general/t8_step5_element_data_c_interface.c:230]:
```suggestion
  /* Copy the element's volumes from our data array to the output array. */
```

File: tutorials/general/t8_step6_stencil.cxx [hunk 14]
```diff
@@ -361,7 +361,7 @@ t8_step6_output_data_to_vtu (t8_forest_t forest,
   /* The name of the field as should be written to the file. */
   strcpy (vtk_data[0].description, "height");
   vtk_data[0].data = heights;
-  /* Copy the elment's volumes from our data array to the output array. */
+  /* Copy the 's volumes from our data array to the output array. */
```
Comment (sandro-elsweijer) [inline tutorials/general/t8_step6_stencil.cxx:364]:
```suggestion
  /* Copy the element's volumes from our data array to the output array. */
```

File: tutorials/general/t8_step7_interpolation.cxx [hunk 15]
```diff
@@ -255,7 +255,7 @@ t8_write_vtu (t8_forest_t forest,
   vtk_data.type = T8_VTK_SCALAR;
   strcpy (vtk_data.description, "Element own data");
   vtk_data.data = element_data;
-  /* Copy the elment's data from the data array to the output array. */
+  /* Copy the 's data from the data array to the output array. */
```
Comment (sandro-elsweijer) [inline tutorials/general/t8_step7_interpolation.cxx:258]:
```suggestion
  /* Copy the element's data from the data array to the output array. */
```

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
We discussed the spell checking exceptions you added to the spell checker and decided, that we will make a new release shortly after merging this PR. Therefore, it would be nice if you could remove the exceptions and run the spell checking again :)

**Comment by sandro-elsweijer** [comment]
A quick question about the spell checker: Since the newly implemented workflow did not run yet, I suppose you ran the spell check locally? Which commands did you use for that?

**Comment by ranocha** [comment]
Thanks for the quick review! I applied your suggestions.

> @ranocha, do you know why the spell checker wants to replace `occured` with `diagonal`?

Maybe a copy-pasta error?

**Comment by ranocha** [comment]
> A quick question about the spell checker: Since the newly implemented workflow did not run yet, I suppose you ran the spell check locally? Which commands did you use for that?

`typos`. See https://github.com/crate-ci/typos#install

**Review by sandro-elsweijer — APPROVED** [review]

**Comment by ranocha** [comment]
Thanks for your review and the fixes!

**Pull request merged**